lib/client.js

```diff
@@ -122,9 +143,19 @@ class Client {
 
       log.info('GET: ' + requestOptions.uri)
       request(requestOptions, (error, resp, body) => {
+        // Handle low level, non-http error:
         if (error) reject(error)
-        else if (!body || !body.data) reject('Invalid response: for ' + url, body)
-        else {
+
+        // Handle token expiration:
+        else if (body && body.statusCode === 401) {
```

Comment:
Would it be better to check the status code from the HTTP response itself (`resp`)? Trusting the actual status code might be a little more trustworthy and make it a little less brittle.